### PR TITLE
Fix bug between snake casing and caps

### PR DESCRIPTION
**Pull request by quintenpalmer**
If snake casing and requesting all caps there would be a leading _
prefix, which was undesired. This resolves that while keeping other
functionality the same.

### CODE REVIEW

File: jsonenums.go [hunk 1]
```diff
@@ -162,23 +162,21 @@ func main() {
...
 					log.Fatalf("Error removing prefix: %v", err)
 				}
-				cammelSnakePairs[i].SnakeRep = prefixRemoved
+			}
+
+			cammelSnakePairs[i].CammelRep = rawValue
+			if exportSnakeCaseJSON != nil && *exportSnakeCaseJSON {
+				cammelSnakePairs[i].SnakeRep = ToSnake(value)
+			} else {
+				cammelSnakePairs[i].SnakeRep = value
```
Comment (hchang-clypd) [inline jsonenums.go:179]: how come no test updates / changes?

Comment (quintenpalmer) [reply jsonenums.go:179]: The pattern we have now is example code, included it here: https://github.com/clyphub/jsonenums/pull/6/commits/f89bac93f94393943203f27425fd0d9604661501

We need to break down this main to get better automated tests, here is the story I wrote: https://clypdinc.atlassian.net/browse/MAIN-17734

edit: @hchang-clypd ping

Comment (hchang-clypd) [reply jsonenums.go:179]: sadfaceok

**Comment by hchang-clypd** [comment]
👍

**Pull request merged**